src/input.h

```diff
@@ -375,6 +380,10 @@ class input_context
 
         input_context() : registered_any_input( false ), category( "default" ),
             handling_coordinate_input( false ) {
+#ifdef GSI
```

Comment:
All of these blocks need to be invocations of gsi methods that contain appropriate ifdefs.